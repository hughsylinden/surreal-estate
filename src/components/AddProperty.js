import React from "react";

const AddProperty = () => {
  return (
    <div>
      <h1>add property</h1>
    </div>
  );
};

export default AddProperty;
