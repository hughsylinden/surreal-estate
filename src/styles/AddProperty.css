.AddProperty{
  background-color: white;
}

label{
  display:flex;
  justify-content: space-between;
  margin:10px;
}
input{
  width: 150px;
}

.addproperty__div{
  display:flex;
  flex-direction: column;
  width:300px;
}