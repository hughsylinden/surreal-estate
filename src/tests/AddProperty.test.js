import { render, screen } from "@testing-library/react";
// import renderer from "react-test-renderer";
import React from "react";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  it("renders AddProperty Page", () => {
    const { asFragment } = render(<AddProperty />);
    expect(asFragment).toMatchSnapshot();
  });

  it("renders correct text output", () => {
    /*   const dummyProperty = {
    title: "dummyTitle",
    bedrooms: 1,
    bathrooms: 2,
    price: 3,
    city: "dummyCity",
    type: "dummyType",
    email: "dummyEmail",
  }; */

    render(<AddProperty />);
    expect(screen.getByText("title:")).toBeInTheDocument();
    expect(screen.getByText("bedrooms:")).toBeInTheDocument();
    expect(screen.getByText("bathrooms:")).toBeInTheDocument();
    expect(screen.getByText("price:")).toBeInTheDocument();
    expect(screen.getByText("email:")).toBeInTheDocument();
    expect(screen.getByText("city:")).toBeInTheDocument();
    expect(screen.getByText("type:")).toBeInTheDocument();
  });

  /*   it("pressing button with no input renders alert component", async () => {
    const handleAddProperty = jest.fn();
    render(<AddProperty />);
    const button = screen.getByText("Add");
    await fireEvent.click(button);
    expect(handleAddProperty).toBeCalledTimes(1);
  }); */
});
