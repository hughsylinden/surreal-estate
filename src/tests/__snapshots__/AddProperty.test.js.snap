// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty pressing button with no input renders alert component 1`] = `
<div
  className="AddProperty"
  onSubmit={[Function]}
>
  
  <form>
    <div
      className="addproperty__div"
    >
      <label
        className="addproperty__label"
        htmlFor="title"
      >
        title:
        <input
          id="title"
          name="title"
          onChange={[Function]}
          placeholder="John Doe"
          value=""
        />
      </label>
      <label
        className="addproperty__label"
        htmlFor="bedrooms"
      >
        bedrooms:
        <input
          id="bedrooms"
          name="bedrooms"
          onChange={[Function]}
          placeholder="1-10"
        />
      </label>
      <label
        className="addproperty__label"
        htmlFor="bathrooms"
      >
        bathrooms:
        <input
          id="bathrooms"
          name="bathrooms"
          onChange={[Function]}
          placeholder="1-10"
        />
      </label>
      <label
        className="addproperty__label"
        htmlFor="price"
      >
        price:
        <input
          id="price"
          name="price"
          onChange={[Function]}
          placeholder="0-2000000Â£"
        />
      </label>
      <label
        className="addproperty__label"
        htmlFor="email"
      >
        email:
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="johndoe@email.com"
        />
      </label>
      <label
        className="addproperty__label"
        htmlFor="city"
      >
        city:
        <select
          id="city"
          name="city"
          onChange={[Function]}
          value="Manchester"
        >
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Sheffield"
          >
            Sheffield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </label>
      <label
        className="addproperty__label"
        htmlFor="type"
      >
        type:
        <select
          id="type"
          name="type"
          onChange={[Function]}
        >
          <option
            value="Flat"
          >
            Flat
          </option>
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-Detached
          </option>
          <option
            value="End of Terrace"
          >
            End of Terrace
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
        </select>
      </label>
      <button
        type="submit"
      >
        Add
      </button>
    </div>
  </form>
</div>
`;

exports[`AddProperty renders AddProperty Page 1`] = `[Function]`;
